File: web_frontend/src/api/traces.ts
import { apiGet } from "./client";

export type TraceSummary = {
  time: string;
  trace_id: string;
  shop_id: string;
  buyer_id: string;
  session_id: string;
  intent: string;
  domain: string | null;
  action: string;
  rag_hit_count: number;
  final_status: string;
  latency_ms: number;
  no_send: boolean;
};

export function getTraces() {
  return apiGet<{ items: TraceSummary[]; total: number }>("/api/traces");
}

export function getTrace(traceId: string) {
  return apiGet<Record<string, unknown>>(`/api/traces/${traceId}`);
}

File: web_api/schemas/traces.py
from pydantic import BaseModel


class TraceSummary(BaseModel):
    time: str
    trace_id: str
    shop_id: str
    buyer_id: str
    session_id: str
    intent: str
    domain: str | None
    action: str
    rag_hit_count: int
    guardrail_status: str
    answer_status: str
    final_status: str
    latency_ms: int
    no_send: bool
